controller/iteration.go

```diff
@@ -53,26 +68,35 @@ func (c *IterationController) CreateChild(ctx *app.CreateChildIterationContext)
 	if err != nil {
 		return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
 	}
-
-	return application.Transactional(c.db, func(appl application.Application) error {
-
-		parent, err := appl.Iterations().Load(ctx, parentID)
+	var parent *iteration.Iteration
+	var sp *space.Space
+	err = application.Transactional(c.db, func(appl application.Application) error {
+		parent, err = appl.Iterations().Load(ctx, parentID)
 		if err != nil {
-			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+			return err
 		}
-		s, err := appl.Spaces().Load(ctx, parent.SpaceID)
+		sp, err = appl.Spaces().Load(ctx, parent.SpaceID)
 		if err != nil {
-			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
-		}
-		if !uuid.Equal(*currentUser, s.OwnerID) {
-			log.Warn(ctx, map[string]interface{}{
-				"space_id":     s.ID,
-				"space_owner":  s.OwnerID,
-				"current_user": *currentUser,
-			}, "user is not the space owner")
-			return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not the space owner"))
+			return jsonapi.JSONErrorResponse(ctx, err)
 		}
-
+		return nil
+	})
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+	authorized, spaceOwner, err := verifyUser(ctx, *currentUser, sp)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+	}
+	if !authorized && !spaceOwner {
```

Comment:
If `!authorized` is `true`, we should return 401 instead of 403, right?

Reply:
please see https://github.com/fabric8-services/fabric8-wit/pull/1783/commits/f92e75707c4e9e632abb890011e820acb685271c